src/components/text-editor/TextEditor.tsx

```diff
@@ -0,0 +1,158 @@
+// Custon ESlint rule overide fixes a conflict between ESlint and Prettier
+// rules on a certain weird indentation edge case on line 74-76. on var input
+/* eslint @typescript-eslint/indent: 0 */
+import React, { useState } from "react";
+import styled, { AnyStyledComponent } from "styled-components";
+
+const Main: AnyStyledComponent = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  align-items: center;
+  height: 100%;
+  width: 100%;
+  text-align: center;
+`;
+
+const Terminal: AnyStyledComponent = styled.form`
+  background-color: #fff;
+  border-radius: 5px;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  height: 90%;
+  width: 70%;
+`;
+
+const Text: AnyStyledComponent = styled.textarea`
+  margin: 0;
+  border-radius: 5px;
+  border-style: none;
+  height: 75%;
+  margin: 0 0 10px 0;
+  resize: none;
+  width: 80%;
+  font-size: 16px;
+
+  :focus {
+    outline: none;
+  }
+`;
+
+const Button: AnyStyledComponent = styled.button`
+  background-color: #90c577;
+  border-radius: 5px;
+  border-style: none;
+  height: 43px;
+  width: 98px;
+  font-size: 18px;
+
+  @media only screen and (max-width: 800px) {
+    bottom: 60px;
+  }
+`;
+
+const SuggestBox: AnyStyledComponent = styled.div`
+  height: 50px;
+  margin: 0 0 10px 0;
+  max-width: 80%;
+
+  h1 {
+    font-size: 18px;
+    font-weight: normal;
+  }
+`;
+
+const TextEditor: React.FC = (): JSX.Element => {
+  const [suggestion, setSuggestion] = useState<string>("");
+
+  // TODO: Import external array containing the actual keywords
+  // These are the words that we want to suggest to the user
+  const words: string[] = ["drone", "python", "code"];
+
+  // We are using uncontrolled method of creating Forms in React, so we go
+  // ahead and set this variable to the content inside of the terminal
+  const input: React.RefObject<HTMLTextAreaElement> = React.createRef<
+    HTMLTextAreaElement
+  >();
+
+  const removeSuggestion = (): void => {
+    setSuggestion("");
+  };
+
+  // Uses selectionStart to detect where the user's cursor position is
+  // and offers a suggestion based on what the user has typed since then
+  const autoCompleteCheck = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ): void => {
+    let text: string = event.target.value;
```

Comment:
I would prefer you control the react component and have the value in state

Reply:
https://reactjs.org/docs/forms.html scroll to "The textarea Tag"

Reply:
Basically just have the onchange be a function that simple updates the state and then do a useeffect hook to check for autocomplete